packages/cli/src/imagerenderer.ts

```diff
@@ -0,0 +1,149 @@
+import sharp from "sharp";
+
+export type TerminalType = "kitty" | "iterm2" | "none";
+
+const KITTY_IDENTIFIERS: string[] = [
+  "kitty",
+  "wezterm",
+  "konsole",
+  "warp",
+  "wayst",
+  "st",
+];
+
+type KittyCommand = Record<string, string | number>;
+
+export function detectTerminalCapabilities(): TerminalType {
+  const termProgram = (Deno.env.get("TERM_PROGRAM") || "").toLowerCase();
+
+  for (const id of KITTY_IDENTIFIERS) {
+    if (termProgram === id) {
+      return "kitty";
+    }
+  }
+
+  if (termProgram === "iterm.app") {
+    return "iterm2";
+  }
+
+  return "none";
+}
+
+function serializeGrCommand(
+  cmd: KittyCommand,
+  payload?: string,
+): Uint8Array {
+  const cmdString = Object.entries(cmd)
+    .map(([k, v]) => `${k}=${v}`)
+    .join(",");
+
+  const encoder = new TextEncoder();
+  const parts: Uint8Array[] = [];
+
+  parts.push(encoder.encode("\x1b_G"));
+  parts.push(encoder.encode(cmdString));
+
+  if (payload) {
+    parts.push(encoder.encode(";"));
+    parts.push(encoder.encode(payload));
+  }
+
+  parts.push(encoder.encode("\x1b\\"));
+
+  const totalLength = parts.reduce((sum, part) => sum + part.length, 0);
+  const result = new Uint8Array(totalLength);
+  let offset = 0;
+
+  for (const part of parts) {
+    result.set(part, offset);
+    offset += part.length;
+  }
+
+  return result;
+}
+
+export async function renderImageKitty(
+  imagePath: string,
+  cmd: KittyCommand,
+): Promise<void> {
+  const data = await Deno.readFile(imagePath);
+  const base64Data = btoa(String.fromCharCode(...data));
+  let remaining = base64Data;
+  let isFirst = true;
+
+  while (remaining.length > 0) {
+    const chunk = remaining.slice(0, 4096);
+    remaining = remaining.slice(4096);
+
+    const chunkCmd = {
+      ...(isFirst ? cmd : {}),
+      m: remaining.length > 0 ? 1 : 0, // The required 'm' property
+    };
+    chunkCmd.m = remaining.length > 0 ? 1 : 0;
+    const command = serializeGrCommand(chunkCmd, chunk);
+
+    Deno.stdout.writeSync(command);
+
+    isFirst = false;
+  }
+}
+
+export async function renderImageITerm2(
+  imagePath: string,
+): Promise<void> {
+  const imageData = await Deno.readFile(imagePath);
+  const base64Data = btoa(String.fromCharCode(...imageData));
+
+  const encoder = new TextEncoder();
+  const command = encoder.encode(
+    `\x1b]1337;File=inline=1preserveAspectRatio=1:${base64Data}\x07\n`,
+  );
+  Deno.stdout.writeSync(command);
+}
+
+export async function downloadImage(url: string): Promise<string | null> {
+  try {
+    const response = await fetch(url);
+    const imageData = new Uint8Array(await response.arrayBuffer());
+    const tempDir = Deno.env.get("TMPDIR") || Deno.env.get("TMP") || "/tmp";
+    const filename = `terminal_image_${Date.now()}_${crypto.randomUUID()}`;
+    const extension = new URL(url).pathname.split(".").pop() || "jpg";
+    const tempPath = `${tempDir}/${filename}.${extension}`;
```

Comment:
You could utilize [`Deno.makeTempFile()`](https://docs.deno.com/api/deno/~/Deno.makeTempFile) API instead.